### DISCUSSION TITLE
Import of existing json data sources and the LOCATION field

### DISCUSSION THREAD

**Discussion by petermixt** [code4recovery/12-step-meeting-list#854] [General]
### Contact Details

petermixt@gmail.com

### What happened?

So I made discovery about the "Meeting Finder" software and the LOCATION field today. Since a meeting can be saved as a DRAFT without a location the following situation can occur. **Not sure if this is a bug.**

The "Meeting Finder" creates the JSON file to “print” the UI's/meeting/pages. The local JSON reads just fine as it does not see “DRAFT” entries.  However, if the JSON file is being imported from another website as below...

![Screenshot 2022-07-04 at 12-59-32 Import   Settings ‹ CMA Roundups and More — WordPress](https://user-images.githubusercontent.com/38595503/177213145-33c5b680-ffc3-41a9-973b-3ac69737bbfc.png)

... and if the user runs a database refresh with any of the originating directory's meeting's LOCATION fields left **blank** -- the meeting system on the receiving end gives the invalid response error as shown below.

![Screenshot 2022-07-04 at 11-17-40 Import   Settings ‹ CMA Roundups and More — WordPress](https://user-images.githubusercontent.com/38595503/177213175-0e4c9cfe-87e4-44dc-927a-c26873004d4a.png)

A blank LOCATION field in the originating directory was been saved as a draft.  Even though the meeting was only a "draft", the receiving database crashes, returning the invalid response error as the import tries to update. Adding the originating directory's meeting's missing LOCATION fixes the invalid response error. 

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

### What browsers are you seeing the problem on?

Firefox

**Comment by tech2serve** [comment]
Let's investigate this before creating an issue. 

Are you saying that on site A, a meeting in draft status, was included in a feed to site b?

**Comment by petermixt** [comment]
No,  Site A has a meeting with no location listed in meeting's "location filed", just and address, then saved as a DRAFT... Site B import crashed giving the invalid response.
When Site A meetings "location filed" was filled with some data, Site B import did not crash.

**Reply by tech2serve** [reply]
Interesting. I'm not sure why the JSON feed from Site A should include any meetings that are draft to begin with. 

Can you share a link to Site A?

**Comment by petermixt** [comment]
Discovered this only because our Registrar volunteer did not enter a new listing properly.

**Comment by petermixt** [comment]
Yes this seems to have nothing to do with the JSON file from Site A.  More something at Site B.  We have been experiencing a large DOS attack at Site B, so until this is all stable I am putting a cap on it since it is a text site and nothing for the public.

**Comment by petermixt** [comment]
Further this issue appeared when set to "Legacy" as:
![Screenshot-(54)](https://user-images.githubusercontent.com/38595503/178612081-d70edf18-4a29-4582-abb8-66f067070743.jpg)

When changing to "Beta" the issue cleared .   hmmmmmmm

**Reply by tim-rohrer** [reply]
What version of TSML is being used?

**Reply by petermixt** [reply]
Version 3.14.5 on both sites.

**Reply by petermixt** [reply]
Site A developed the inability to finish an entry. the Legacy version used failed at the address field.  Changing site A to Beta restored the address field function.  Could this be a change at Google?

**Reply by tim-rohrer** [reply]
This particular issue is related to our Google account, and it has now been resolved.

Geocoding through Legacy should function as expected again.
